stdlib/public/core/Availability.swift

```diff
@@ -45,3 +45,52 @@ public func _stdlib_isOSVersionAtLeast(
   return false._value
 #endif
 }
+
+/// Returns the version number for the Swift Standard Library that the code
+/// calling this was originally compiled with.
+///
+/// The version number is an arbitrary pair of integers, with lexicographical
+/// ordering. Version numbers of (logically) successive stdlib releases form a
+/// monotonically increasing sequence; i.e., versions should not decrease, but
+/// they are allowed to stay the same.
+///
+/// The two integer components are not intended to correspond to major or minor
+/// versions in a semantic versioning scheme. Neither do they correlate with
+/// toolchain or OS version numbers.
+public var _stdlibStaticVersion: (Int, Int) {
```

Comment:
Still a little confused. Why is this public?

Reply:
It allows tests to easily determine if they are back/forward deployed. Binaries that are intimately tied to a particular revision of the stdlib and should not be separately back/forward deployed (e.g. libswiftFoundation) could use it for sanity checks.

This is not strictly necessary, but it seems like a cheap addition.

Reply:
FWIW, dyld provides two similar version numbers through `NSVersionOfRunTimeLibrary` and `NSVersionOfLinkTimeLibrary`. 

(The link-time version queries the linkage of the main executable, not the current object, so it's different(ly useful) than the static version number implemented here.)